src/osgViewer/View.cpp

```diff
@@ -1130,6 +1130,23 @@ void View::removeDevice(osgGA::Device* eventSource)
     }
 }
 
+bool View::requiresUpdateSceneGraph() const
+{
+    // check if the database pager needs to update the scene
+    if (getDatabasePager()->requiresUpdateSceneGraph() || getDatabasePager()->getRequestsInProgress()) return true;
+
+    // check if there are camera update callbacks
+    if (_camera->getUpdateCallback()) return true;
+
+    // check if there are node update callbacks
+    if (getSceneData() != 0)
+    {
+        if (getSceneData()->getUpdateCallback() || (getSceneData()->getNumChildrenRequiringUpdateTraversal()>0)) return true;
```

Comment:
the whole logic could be moved to a new SceneData::requiresUpdateSceneGraph() method ?